documentation/PreemptiveProvableAssertions.md

```
@@ -0,0 +1,448 @@
+
+## Interdependent L2 transactions
+
+One category of state that the sequencer can predict is the L2 state within the publication that they are constructing. Having decided to respect some constraint about future state, they can assert that claim immediately so L2 contracts can rely on it.
+
+<p align="center"><img src="./provable_assertion_images.7.png"/></p>
+
+This could simplify interdependent transactions. For example, consider a literal Prisoner's Dilemma contract.
+
+```solidity
+contract PrisonersDilemma is IPrisonersDilemma {
+	const uint256 COOPERATE = 1;
```

markdownlint-cli2 (0.17.2)

Hard tabs
Column: 1

(MD010, no-hard-tabs)